surefire-api/src/main/java/org/apache/maven/surefire/api/report/LegacyPojoStackTraceWriter.java

```diff
@@ -81,19 +80,26 @@ public String smartTrimmedStackTrace()
         result.append( "#" );
         result.append( testMethod );
         SafeThrowable throwable = getThrowable();
-        if ( throwable.getTarget() instanceof AssertionError )
-        {
-            result.append( " " );
-            result.append( getTruncatedMessage( throwable.getMessage(), MAX_LINE_LENGTH - result.length() ) );
-        }
-        else
+        Throwable target = throwable.getTarget();
+
+        if ( target != null )
         {
-            Throwable target = throwable.getTarget();
-            if ( target != null )
+
+            final String excClassName = target.getClass().getName();
+
+            if ( ! ( AssertionError.class.isInstance( target )
+                    || excClassName.endsWith( ".AssertionFailedError" )
+                    || excClassName.endsWith( ".ComparisonFailure" )
+                    || excClassName.startsWith( "org.opentest4j." ) ) )
+            {
+                result.append( ' ' )
+                      .append( target.getClass().getSimpleName() );
+            }
+            final String msg = throwable.getMessage();
+            if ( isNotEmpty( msg ) )
             {
-                result.append( " " );
-                result.append( target.getClass().getSimpleName() );
-                result.append( getTruncatedMessage( throwable.getMessage(), MAX_LINE_LENGTH - result.length() ) );
+                result.append( ' ' )
+                      .append( msg.split( "\r?\n" )[0] );
```

Comment:
I know that the `.split( "\r?\n" )[0]` was my request but this part of code will break the nackwards compatibility. Pls use only `.append( msg );`, thx
And I wahe additionalt question. Is the code between the lines 90 and 103 the same in the class `SmartStackTraceParser` - see the lines 133 - 146? Is it possible to share this code between classes?
Has the method `toMinimalThrowableMiniMessage()` the same code in both classes?

Reply:
Hi @Tibor17

> I know that the `.split( "\r?\n" )[0]` was my request but this part of code will break the nackwards compatibility.

In what way would this break backward compatibility?

Surefire has always kept each failure on one line on the output. This behaviour is still true.

If your concern is that important info is contained on lines 2..n of the error/failure message, we could replace line separators by space and keep the complete message.
```suggestion
                      .append( msg.replaceAll( "\r?\n", " " ) );
```

Reply:
The mesage was always printed completely.
If you compare witht he old code, you will see only `msg`.
The point of the issue is to remove the function which takes first 77 chars, right?

Reply:
> The point of the issue is to remove the function which takes first 77 chars, right?

Correct.
Or to put in more general terms: output enough of the error message that it is always clear and eases troubleshooting.
Your suggestion to print only the first line is a good compromise I think.
To have multi-line messages in exception is unusual (if not an anti-pattern). I cannot think of many examples other than jdbc drivers maybe that tend to be quite verbose in the exception message.

Reply:
The Spark/Groovy is having such assertion statements which have excellent benefit with multiline message for user.
I would not like to break the users since they rely on complete message.
Thx for your work and all your efforts!

Reply:
Fine with me. I'll close this pull request then?